drivers/can/can_mcux_flexcan.c

```diff
@@ -633,7 +672,37 @@ static int mcux_flexcan_init(const struct device *dev)
 		k_sem_init(&data->tx_cbs[i].done, 0, 1);
 	}
 
-	err = mcux_flexcan_configure(dev, CAN_NORMAL_MODE, 0);
+	timing.sjw = config->sjw;
+	if (config->sample_point) {
+		err = can_calc_timing(dev, &timing, config->bitrate,
+				      config->sample_point);
+		if (err == -EINVAL) {
+			LOG_ERR("Can't find timing for given param");
+			return -EIO;
+		}
+		LOG_DBG("Presc: %d, Seg1S1: %d, Seg2: %d",
+			timing.prescaler, timing.phase_seg1, timing.phase_seg2);
+		LOG_DBG("Sample-point err : %d", err);
+	} else if (config->phase_seg1) {
+		timing.prop_seg = config->prop_seg;
+		timing.phase_seg1 = config->phase_seg1;
+		timing.phase_seg2 = config->phase_seg2;
+		err = can_calc_prescaler(dev, &timing, config->bitrate);
+		if (err) {
+			LOG_WRN("Bitrate error: %d", err);
+		}
+	} else {
+		LOG_ERR("Timing not configured");
+		return -EINVAL;
+	}
+
+	data->timing.preDivider = timing.prescaler;
+	data->timing.rJumpwidth = timing.sjw;
+	data->timing.phaseSeg1 = timing.phase_seg1;
+	data->timing.phaseSeg2 = timing.phase_seg2;
+	data->timing.propSeg = timing.prop_seg;
```

Comment:
Maybe call `mcux_flexcan_set_timing(dev, timing)` instead?

Reply:
That would call the set bitrate on an uninitialized controller. I'd avoid that

Reply:
Ok, makes sense.